galley/pkg/config/resource/name.go

```diff
@@ -28,7 +32,23 @@ func NewName(namespace, local string) Name {
 	return Name{string: namespace + "/" + local}
 }
 
-// String inteface implementation.
+// NewFullName returns a given name as a resource Name
+func NewFullName(name string) (Name, error) {
+	if name == "" {
+		return Name{string: ""}, errors.New("invalid name: can not be empty")
+	}
+	if strings.Contains(name, "/") {
+		segments := strings.Split(name, "/")
```

Comment:
This will cause allocations. Any chance we can write this in non-allocating way? Something like:
```
func() {
  begin := 0
  for i, r := range name {
    if r == '/' {
      if begin == i {
	return err
      }
      begin = i + 1
    }
  }
  if begin == len(name) {
	return err
  }
  return nil
}
```